lib/excon/socket.rb

```diff
@@ -324,5 +344,16 @@ def unpacked_sockaddr
         raise
       end
     end
+
+    # Checks whether we have exceeded the request deadline.
+    # Returns the remaining time in seconds until we reach the deadline.
+    def check_deadline!
```

Comment:
It might be clearer to name this based on our intention, rather than what it does. So rather than saying it checks the deadline, maybe something like `request_time_remaining` would make it more obvious?

Reply:
That name makes sense to me. I renamed to `request_time_remaining` and updated the doc comment to make it clearer that we are checking the time remaining until we hit the request's timeout.